contracts/account/modules/ERC7579MultisigConfirmation.sol

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.27;
+
+import {ERC7913Utils} from "../../utils/cryptography/ERC7913Utils.sol";
+import {ERC7579Multisig} from "./ERC7579Multisig.sol";
+import {EIP712} from "@openzeppelin/contracts/utils/cryptography/EIP712.sol";
+
+/**
+ * @dev Extension of {ERC7579Multisig} that requires explicit confirmation signatures
+ * from new signers when they are being added to the multisig.
+ *
+ * This module ensures that only willing participants can be added as signers to a
+ * multisig by requiring each new signer to provide a valid signature confirming their
+ * consent to be added. Each signer must sign an EIP-712 message to confirm their addition.
+ *
+ * TIP: Use this module to ensure that all guardians in a social recovery or multisig setup have
+ * explicitly agreed to their roles.
+ */
+abstract contract ERC7579MultisigConfirmation is ERC7579Multisig, EIP712 {
+    bytes32 private constant MULTISIG_CONFIRMATION = keccak256("MultisigConfirmation(address account,address module)");
```

Comment:
Would it make sense to add a nonce value? Confirmations could be reused in this way, but I don't see why that's an issue if you want to only validate that the signer does exists and can sign. Perhaps expiration?

Reply:
Try to come up with a concrete opinion. Based on your comments, it seems that 1) it's no a big deal if the confirmation is perpetual and 2) contracts may change, so I'd argue that you're suggesting a configurable optional expiration (e.g. `type().max == expiration`). I would agree though.

Reply:
Added expiration encoded in the signature field